File: app/src/main/java/mydebts/android/app/data/internal/CursorToParticipant.java
package mydebts.android.app.data.internal;

import android.database.Cursor;

import io.reactivex.functions.Function;
import mydebts.android.app.data.db.ParticipantContract;
import mydebts.android.app.data.model.Participant;

class CursorToParticipant implements Function<Cursor, Participant> {
    private static final CursorToEvent CURSOR_TO_EVENT = new CursorToEvent();
    private static final CursorToPerson CURSOR_TO_PERSON = new CursorToPerson();

    @Override
    public Participant apply(Cursor cursor) throws Exception {
        return Participant.builder()
                .id(cursor.getLong(cursor.getColumnIndex(ParticipantContract._ID)))
                .event(CURSOR_TO_EVENT.apply(cursor))
                .person(CURSOR_TO_PERSON.apply(cursor))
                .debt(cursor.getDouble(cursor.getColumnIndex(ParticipantContract.COLUMN_DEBT)))
                .build();
    }
}

File: app/src/main/java/mydebts/android/app/data/internal/CursorToParticipant.kt
package mydebts.android.app.data.internal

import android.database.Cursor

import io.reactivex.functions.Function
import mydebts.android.app.data.db.ParticipantContract
import mydebts.android.app.data.model.Participant

internal class CursorToParticipant : Function<Cursor, Participant> {

    @Throws(Exception::class)
    override fun apply(cursor: Cursor): Participant {
        return Participant(cursor.getLong(cursor.getColumnIndex(ParticipantContract.COLUMN_ID)),
                CURSOR_TO_EVENT.apply(cursor),
                CURSOR_TO_PERSON.apply(cursor),
                cursor.getDouble(cursor.getColumnIndex(ParticipantContract.COLUMN_DEBT)))
    }

    companion object {
        private val CURSOR_TO_EVENT = CursorToEvent()
        private val CURSOR_TO_PERSON = CursorToPerson()
    }
}
